src/core/Akka.Remote/Transport/TransportAdapters.cs

```diff
@@ -66,11 +66,16 @@ public TransportAdapters(ExtendedActorSystem system)
             {
                 try
                 {
-                    var adapterTypeName = Type.GetType(adapter.Value);
+                    var adapterTypeName = Type.GetType(adapter.Value, true);
                     // ReSharper disable once AssignNullToNotNullAttribute
                     var newAdapter = (ITransportAdapterProvider)Activator.CreateInstance(adapterTypeName);
```

Comment:
should check for ITransportAdapterProvider compat or throw configuration exception